File: week3/assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int numContent = 0;
        int child = 0;
        int cookie = 0;
        while (cookie < s.length && child < g.length) {
            if (g[child] <= s[cookie]) {
                child++;
                numContent++;
            }
            cookie++;
        }
        return numContent;
    }
}

File: 2026/week3/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        i = 0
        j = 0
        g.sort()
        s.sort()
        while i < len(g) and j < len(s):
            if s[j] >= g[i]:
                i += 1
            j += 1
        return i